java-checks/src/main/java/org/sonar/java/checks/spring/NullableInjectedFieldsHaveDefaultValueCheck.java

```diff
@@ -0,0 +1,150 @@
+      .forEach(trees ->
+        QuickFixHelper.newIssue(context)
+          .forRule(this)
+          .onTree(trees.valueAnnotation)
+          .withMessage("Provide a default null value for this field.")
+          .withSecondaries(new JavaFileScannerContext.Location("The nullable annotation", trees.nullableAnnotation))
+          .withQuickFixes(() -> computeQuickFix(trees.valueAnnotation))
+          .report()
+      );
+  }
+
+  private static List<JavaQuickFix> computeQuickFix(AnnotationTree annotation) {
+    return extractLiteralTree(annotation.arguments().get(0)).map(literalTree -> {
+      // Compute replacement
+      String currentValue = literalTree.value();
+      String replacementValue = currentValue.substring(0, currentValue.lastIndexOf('}'))
+        + ":#{null}}"
+        + "\"";
+      // Insert replacement
+      return List.of(
+        JavaQuickFix.newQuickFix("Set null as default value")
+          .addTextEdit(JavaTextEdit.replaceTree(literalTree.token(), replacementValue))
+          .build()
+      );
+    }).orElse(List.of());
+  }
+
+  /**
+   * Maps a class member to a {@link AnnotationsOfInterest} if it is a field annotated as nullable and injected with a Spring value annotation.
+   *
+   * @param member A class member (eg: method, field, ...)
+   * @return An Optional with a {@link AnnotationsOfInterest} if the member matches. Optional.empty() otherwise.
+   */
+  private static Optional<AnnotationsOfInterest> mapToTreesOfInterest(Tree member) {
+    if (!member.is(Tree.Kind.VARIABLE)) {
+      return Optional.empty();
+    }
+    var field = (VariableTree) member;
+    Optional<AnnotationTree> valueAnnotation = getValueAnnotationWithoutDefault(field);
+    if (valueAnnotation.isEmpty()) {
+      return Optional.empty();
+    }
+    Optional<AnnotationTree> nullableAnnotation = getNullableAnnotation(field);
+    return nullableAnnotation.map(annotationTree -> new AnnotationsOfInterest(valueAnnotation.get(), annotationTree));
+  }
+
+  private static Optional<AnnotationTree> getValueAnnotationWithoutDefault(VariableTree field) {
+    return field.modifiers().annotations().stream()
+      .filter(annotation -> annotation.symbolType().is(VALUE_ANNOTATION) && !hasDefaultValue(annotation))
+      .findFirst();
+  }
+
+  private static Optional <AnnotationTree> getNullableAnnotation(VariableTree field) {
+    SymbolMetadata.NullabilityData nullabilityData = field.symbol().metadata().nullabilityData(SymbolMetadata.NullabilityTarget.FIELD);
+    SymbolMetadata.AnnotationInstance instance = nullabilityData.annotation();
+    if (instance == null) {
+      return Optional.empty();
+    }
+    return Optional.ofNullable(field.symbol().metadata().findAnnotationTree(instance));
+  }
+
+  private static boolean hasDefaultValue(AnnotationTree annotation) {
+    ExpressionTree expression = annotation.arguments().get(0);
+    return extractLiteralValue(expression)
+      .map(argument -> argument.contains(":"))
```

Comment:
I don't think this is quite enough to avoid FPs. Notably, we should probably ignore any real expressions, so things starting with `#`. The following is valid and currently raises issues:
```
@Value("#{\"foo\"}")
```
even more relevant:
```
@Value("#{null}")
```

Reply:
Indeed, the logic is too naive (so is the one for the quick-fix). I will have another try at hardening it